src/utils/api/withAuthorization.ts

```diff
@@ -0,0 +1,28 @@
+import {
+	APIGatewayProxyEvent,
+	APIGatewayProxyHandler,
+	APIGatewayProxyResult,
+	Callback,
+	Context,
+} from 'aws-lambda';
+import { validateAuthorizationToken } from '@utils/api/apiAuth';
+
+export function withAuthorization(handler: APIGatewayProxyHandler): APIGatewayProxyHandler {
+	return async (
+		event: APIGatewayProxyEvent,
+		context: Context,
+		callback: Callback<APIGatewayProxyResult>
+	): Promise<APIGatewayProxyResult> => {
+		const authorizationValidationResult = validateAuthorizationToken(event);
+
+		if (authorizationValidationResult) {
+			return authorizationValidationResult;
+		}
+
+		const result = await handler(event, context, callback);
+		if (result === undefined) {
+			throw new Error('Handler returned undefined');
+		}
```

Comment:
You shouldn't throw here at this point. This is actually a valid response for certain states. You should do a 204 response.